### ISSUE TITLE
String literal union types are validated, but values are not suggested by IDEs

### ISSUE THREAD

**Issue by grantbowering** [Microsoft/TypeScript#12891]
In the 3 IDEs I've tried (VS2015u3, VS2017RC, and VSCode 1.7.2), string literal union type values are detected and underlined in real time if invalid, and the error message lists values that would be valid, but the IDE won't help you type those values or suggest anything. 

**TypeScript Version:**  2.1.4

**Code**

```ts
// A *self-contained* demonstration of the problem follows...
import { Reducer } from 'redux';

export interface CounterState {
    count: number;
}

interface IncrementCountAction { type: 'INCREMENT_COUNT' }
interface DecrementCountAction { type: 'DECREMENT_COUNT' }

export type KnownAction = IncrementCountAction | DecrementCountAction;

export const reducer: Reducer<CounterState> = (state: CounterState, action: KnownAction) => {
    switch (action.type) {
        case 'INCREMENT_CO':
            return { count: state.count + 1 };
        case 'DECREMENT_COUNT':
            return { count: state.count - 1 };
        default:
            const exhaustiveCheck: never = action;
    }

    return state || { count: 0 };
};
```
The value is detected as an error in all IDEs, and the tooltip in all IDEs indicates what values (or the first several until trailing off, if there are many) are valid:

![tsautocomplete1](https://cloud.githubusercontent.com/assets/2152220/21155154/3060eaf2-c12e-11e6-8d33-90dc769680e4.png)

**Expected behavior:**

Auto-completion suggestions should **_include_**, **_and_** be **_limited_** to, the same two possible string values that are indicated in the error.

**Actual behavior:**

![tsautocomplete2](https://cloud.githubusercontent.com/assets/2152220/21155142/27fbcc88-c12e-11e6-9f98-ee44bc2d63e7.png)

VS2015u3 and VS2017RC provide no suggestions.  VSCode technically does provide the suggestion given the sample above, but only because it appears as a literal in the same file and VSCode always does that; if you move the reducer function to a separate file, VSCode will not have a suggestion either.

**Comment by RyanCavanaugh** [comment]
This should be fixed in almost all cases; please log a bug with complete repro steps if you find a case where it isn't working
